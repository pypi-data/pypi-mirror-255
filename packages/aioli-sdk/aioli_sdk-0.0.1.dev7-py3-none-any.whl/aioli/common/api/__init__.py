from aioli.common.api import errors
from aioli.common.api._util import PageOpts
