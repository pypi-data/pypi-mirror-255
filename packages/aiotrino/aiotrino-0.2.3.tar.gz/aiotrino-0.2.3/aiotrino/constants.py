# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from typing import Any, Optional, Text  # NOQA: mypy types

DEFAULT_PORT = 8080
DEFAULT_SOURCE = "aiotrino-python-client"
DEFAULT_CATALOG = None  # type: Optional[Text]
DEFAULT_SCHEMA = None  # type: Optional[Text]
DEFAULT_AUTH = None  # type: Optional[Any]
DEFAULT_MAX_ATTEMPTS = 3
DEFAULT_REQUEST_TIMEOUT = 30.0  # type: float

HTTP = "http"
HTTPS = "https"

URL_STATEMENT_PATH = "/v1/statement"


class TrinoHeaders:
    CATALOG = "X-Trino-Catalog"
    SCHEMA = "X-Trino-Schema"
    SOURCE = "X-Trino-Source"
    USER = "X-Trino-User"
    CLIENT_INFO = "X-Trino-Client-Info"

    SESSION = "X-Trino-Session"
    SET_SESSION = "X-Trino-Set-Session"
    CLEAR_SESSION = "X-Trino-Clear-Session"

    STARTED_TRANSACTION = "X-Trino-Started-Transaction-Id"
    TRANSACTION = "X-Trino-Transaction-Id"

    PREPARED_STATEMENT = 'X-Trino-Prepared-Statement'
    ADDED_PREPARE = 'X-Trino-Added-Prepare'
    DEALLOCATED_PREPARE = 'X-Trino-Deallocated-Prepare'


class PrestoHeaders:
    CATALOG = "X-Presto-Catalog"
    SCHEMA = "X-Presto-Schema"
    SOURCE = "X-Presto-Source"
    USER = "X-Presto-User"
    CLIENT_INFO = "X-Presto-Client-Info"

    SESSION = "X-Presto-Session"
    SET_SESSION = "X-Presto-Set-Session"
    CLEAR_SESSION = "X-Presto-Clear-Session"

    STARTED_TRANSACTION = "X-Presto-Started-Transaction-Id"
    TRANSACTION = "X-Presto-Transaction-Id"

    PREPARED_STATEMENT = 'X-Presto-Prepared-Statement'
    ADDED_PREPARE = 'X-Presto-Added-Prepare'
    DEALLOCATED_PREPARE = 'X-Presto-Deallocated-Prepare'


HEADERS = TrinoHeaders
