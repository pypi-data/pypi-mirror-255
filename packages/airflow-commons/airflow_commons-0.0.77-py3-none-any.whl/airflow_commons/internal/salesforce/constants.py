ASYNC_PATH = "data/v1/async"
SYNC_PATH = "hub/v1"
DATA_PATH = "data/v1"
ASYNC_JOB_RESULT_PATH = "/{request_id}/results"
ASYNC_JOB_STATUS_PATH = "/{request_id}/status"
AUTH_URL = "https://{subdomain}.auth.marketingcloudapis.com/"
DATA_EXTENSION_KEY_PATH = "/dataextensions/key:{key}/rows"
DATA_EVENT_KEY_PATH = "/dataevents/key:{key}/rowset"
CUSTOM_OBJECT_DATA_PATH = "/customobjectdata/key/{key}/rowset"
TOKEN_PATH = "v2/token"
GRANT_TYPE = "client_credentials"
FILTER_QUERY = "$filter={filter}"
