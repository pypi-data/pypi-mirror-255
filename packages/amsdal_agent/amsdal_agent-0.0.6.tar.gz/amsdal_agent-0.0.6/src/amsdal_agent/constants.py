RESTART_ACTION = 'RESTART'
SHUTDOWN_ACTION = 'SHUTDOWN'

ALLOWED_ACTIONS = [
    RESTART_ACTION,
    SHUTDOWN_ACTION,
]
