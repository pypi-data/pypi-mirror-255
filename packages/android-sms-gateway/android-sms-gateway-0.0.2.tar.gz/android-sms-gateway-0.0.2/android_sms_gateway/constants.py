VERSION = "0.0.2"

DEFAULT_URL = "https://sms.capcom.me/api/3rdparty/v1"
