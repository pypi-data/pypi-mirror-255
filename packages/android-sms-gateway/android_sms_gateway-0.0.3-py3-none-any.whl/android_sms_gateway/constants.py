VERSION = "0.0.3"

DEFAULT_URL = "https://sms.capcom.me/api/3rdparty/v1"
