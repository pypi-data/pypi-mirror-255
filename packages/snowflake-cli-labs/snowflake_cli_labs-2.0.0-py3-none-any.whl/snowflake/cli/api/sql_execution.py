from __future__ import annotations

import logging
from contextlib import contextmanager
from functools import cached_property
from io import StringIO
from textwrap import dedent
from typing import Iterable

from click import ClickException
from snowflake.cli.api.cli_global_context import cli_context
from snowflake.connector.cursor import DictCursor, SnowflakeCursor
from snowflake.connector.errors import ProgrammingError


class SqlExecutionMixin:
    def __init__(self):
        pass

    @property
    def _conn(self):
        return cli_context.connection

    @cached_property
    def _log(self):
        return logging.getLogger(__name__)

    def _execute_string(
        self,
        sql_text: str,
        remove_comments: bool = False,
        return_cursors: bool = True,
        cursor_class: SnowflakeCursor = SnowflakeCursor,
        **kwargs,
    ) -> Iterable[SnowflakeCursor]:
        """
        This is a custom implementation of SnowflakeConnection.execute_string that returns generator
        instead of list. In case of executing multiple queries are executed one by one. This mean we can
        access result of previous queries while evaluating next one. For example, we can print the results.
        """
        self._log.debug("Executing %s", sql_text)
        stream = StringIO(sql_text)
        stream_generator = self._conn.execute_stream(
            stream, remove_comments=remove_comments, cursor_class=cursor_class, **kwargs
        )
        return stream_generator if return_cursors else list()

    def _execute_query(self, query: str, **kwargs):
        *_, last_result = self._execute_queries(query, **kwargs)
        return last_result

    def _execute_queries(self, queries: str, **kwargs):
        return list(self._execute_string(dedent(queries), **kwargs))

    @contextmanager
    def use_role(self, new_role: str):
        """
        Switches to a different role for a while, then switches back.
        This is a no-op if the requested role is already active.
        """
        role_result = self._execute_query(
            f"select current_role()", cursor_class=DictCursor
        ).fetchone()
        prev_role = role_result["CURRENT_ROLE()"]
        is_different_role = new_role.lower() != prev_role.lower()
        if is_different_role:
            self._log.debug("Assuming different role: %s", new_role)
            self._execute_query(f"use role {new_role}")
        try:
            yield
        finally:
            if is_different_role:
                self._execute_query(f"use role {prev_role}")

    def _execute_schema_query(self, query: str):
        self.check_database_and_schema()
        return self._execute_query(query)

    def check_database_and_schema(self) -> None:
        database = self._conn.database
        schema = self._conn.schema
        self.check_database_exists(database)
        self.check_schema_exists(database, schema)

    def check_database_exists(self, database: str) -> None:
        if not database:
            raise Exception(
                """
                Database not specified. Please update connection to add `DATABASE` parameter,
                or re-run command using `--dbname` option.
                Use `snow connection list` to list existing connections
                """
            )
        try:
            self._execute_query(f"USE DATABASE {database}")
        except ProgrammingError as e:
            raise Exception(
                f"""
            Exception occurred: {e}. Make sure you have `DATABASE` parameter in connection or `--dbname` option provided
            Use `snow connection list` to list existing connections
            """
            ) from e

    def check_schema_exists(self, database: str, schema: str) -> None:
        if not schema:
            raise Exception(
                """
                Schema not specified. Please update connection to add `SCHEMA` parameter,
                or re-run command using `--schema` option.
                Use `snow connection list` to list existing connections
                """
            )
        try:
            self._execute_query(f"USE {database}.{schema}")
        except ProgrammingError as e:
            raise Exception(
                f"""
            Exception occurred: {e}. Make sure you have `SCHEMA` parameter in connection or `--schema` option provided
            Use `snow connection list` to list existing connections
            """
            ) from e

    def to_fully_qualified_name(self, name: str):
        current_parts = name.split(".")
        if len(current_parts) == 3:
            # already fully qualified name
            return name.upper()

        if not self._conn.database:
            raise ClickException(
                "Default database not specified in connection details."
            )

        if len(current_parts) == 2:
            # we assume this is schema.object
            return f"{self._conn.database}.{name}".upper()

        schema = self._conn.schema or "public"
        return f"{self._conn.database}.{schema}.{name}".upper()
