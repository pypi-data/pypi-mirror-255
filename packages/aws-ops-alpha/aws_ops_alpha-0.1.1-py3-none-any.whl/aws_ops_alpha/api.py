# -*- coding: utf-8 -*-

"""
Usage example:

    >>> import aws_ops_alpha.api as aws_ops_alpha
    >>> aws_ops_alpha.runtime
    ...
"""

# ordered from the module having the least dependencies to the one having most dependencies
from . import constants
from .logger import logger
from .constants import CommonEnvNameEnum
from .constants import EnvVarNameEnum
from .constants import AwsTagNameEnum
from .constants import AwsOpsSemanticBranchEnum
from .env_var import temp_env_var
from .env_var import normalize_env_var_name
from .env_var import get_environment_aws_account_id_in_ci
from .env_var import get_environment_iam_role_arn_in_dev_server
from .env_var import temp_env_var
from .runtime.api import RunTimeGroupEnum
from .runtime.api import RunTimeEnum
from .runtime.api import Runtime
from .runtime.api import runtime
from .multi_env.api import EnvNameValidationError
from .multi_env.api import validate_env_name
from .multi_env.api import BaseEnvNameEnum
from .multi_env.api import env_emoji_mapper
from .multi_env.api import EnvNameEnum
from .multi_env.api import detect_current_env
from .git.api import InvalidSemanticNameError
from .git.api import SemanticBranchRule
from .git.api import GitRepo
from .git.api import extract_semantic_branch_name_for_multi_repo
from .git.api import extract_semantic_branch_name_for_mono_repo
from .git.api import MultiGitRepo
from .git.api import MonoGitRepo
from .boto_ses.api import AbstractBotoSesFactory
from .boto_ses.api import AlphaBotoSesFactory
from .config.api import BaseConfig
from .config.api import BaseEnv
from .config.api import T_BASE_CONFIG
from .config.api import T_BASE_ENV
from .aws_helpers import python_helpers
from .aws_helpers import aws_cdk_helpers
from .aws_helpers import aws_lambda_helpers
from .aws_helpers import aws_ecr_helpers
from .aws_helpers import aws_chalice_helpers

# user may not need the following feature to write their application code
# these features are typically used in CI/CD and devops only
try:
    from .aws_helpers import rich_helpers
except ImportError: # pragma: no cover
    pass

try:
    from .project.api import simple_python_project
    from .project.api import simple_cdk_project
    from .project.api import simple_config_project
    from .project.api import simple_lambda_project
    from .project.api import simple_lbd_container_project
    from .project.api import simple_lbd_agw_chalice_project
except ImportError:  # pragma: no cover
    pass

try:
    from .boostrap import api as boostrap
except ImportError:  # pragma: no cover
    pass
