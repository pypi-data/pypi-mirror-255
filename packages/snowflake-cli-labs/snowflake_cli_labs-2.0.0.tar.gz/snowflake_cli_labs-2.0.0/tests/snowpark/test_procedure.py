import json
from pathlib import Path
from textwrap import dedent
from unittest import mock
from unittest.mock import call

from snowflake.cli.api.constants import ObjectType
from snowflake.connector import ProgrammingError


def test_deploy_function_no_procedure(runner, project_directory):
    with project_directory("empty_project"):
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ],
        )
    assert result.exit_code == 1
    assert "No snowpark project definition found" in result.output


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_procedure(
    mock_object_manager,
    mock_conn,
    runner,
    mock_ctx,
    project_directory,
):

    mock_object_manager.return_value.describe.side_effect = ProgrammingError(
        "does not exist or not authorized"
    )
    ctx = mock_ctx()
    mock_conn.return_value = ctx

    with project_directory("snowpark_procedures") as tmp:
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ]
        )

    assert result.exit_code == 0, result.output
    mock_object_manager.return_value.describe.assert_has_calls(
        [
            call(object_type=str(ObjectType.PROCEDURE), name="procedureName(string)"),
            call(object_type=str(ObjectType.PROCEDURE), name="test()"),
        ]
    )
    assert ctx.get_queries() == [
        "create stage if not exists dev_deployment comment='deployments managed by snowcli'",
        f"put file://{Path(tmp).resolve()}/app.zip @dev_deployment/my_snowpark_project auto_compress=false parallel=4 overwrite=True",
        dedent(
            """\
            create or replace procedure procedureName(name string)
            returns string
            language python
            runtime_version=3.8
            imports=('@dev_deployment/my_snowpark_project/app.zip')
            handler='hello'
            packages=()
            """
        ).strip(),
        dedent(
            """\
            create or replace procedure test()
            returns string
            language python
            runtime_version=3.10
            imports=('@dev_deployment/my_snowpark_project/app.zip')
            handler='test'
            packages=()
            """
        ).strip(),
    ]


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_procedure_with_external_access(
    mock_object_manager,
    mock_conn,
    runner,
    mock_ctx,
    project_directory,
):
    mock_object_manager.return_value.describe.side_effect = ProgrammingError(
        "does not exist or not authorized"
    )
    mock_object_manager.return_value.show.return_value = [
        {"name": "external_1", "type": "EXTERNAL_ACCESS"},
        {"name": "external_2", "type": "EXTERNAL_ACCESS"},
    ]

    ctx = mock_ctx()
    mock_conn.return_value = ctx

    with project_directory("snowpark_procedure_external_access") as project_dir:
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ]
        )

    assert result.exit_code == 0, result.output
    mock_object_manager.return_value.describe.assert_has_calls(
        [
            call(object_type=str(ObjectType.PROCEDURE), name="procedureName(string)"),
        ]
    )
    assert ctx.get_queries() == [
        "create stage if not exists dev_deployment comment='deployments managed by snowcli'",
        f"put file://{Path(project_dir).resolve()}/app.zip @dev_deployment/my_snowpark_project"
        f" auto_compress=false parallel=4 overwrite=True",
        dedent(
            """\
            create or replace procedure procedureName(name string)
            returns string
            language python
            runtime_version=3.8
            imports=('@dev_deployment/my_snowpark_project/app.zip')
            handler='app.hello'
            packages=()
            external_access_integrations=(external_1,external_2)
            secrets=('cred'=cred_name,'other'=other_name)
            """
        ).strip(),
    ]


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_procedure_secrets_without_external_access(
    mock_object_manager,
    mock_conn,
    runner,
    mock_ctx,
    project_directory,
    snapshot,
):
    ctx = mock_ctx()
    mock_conn.return_value = ctx

    mock_object_manager.return_value.show.return_value = [
        {"name": "external_1", "type": "EXTERNAL_ACCESS"},
        {"name": "external_2", "type": "EXTERNAL_ACCESS"},
    ]

    with project_directory("snowpark_procedure_secrets_without_external_access"):
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ],
            catch_exceptions=False,
        )

    assert result.exit_code == 1, result.output
    assert result.output == snapshot


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_procedure_fails_if_integration_does_not_exists(
    mock_object_manager,
    mock_conn,
    runner,
    mock_ctx,
    project_directory,
    snapshot,
):
    ctx = mock_ctx()
    mock_conn.return_value = ctx

    mock_object_manager.return_value.show.return_value = [
        {"name": "external_1", "type": "EXTERNAL_ACCESS"},
    ]

    with project_directory("snowpark_procedure_external_access"):
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ],
            catch_exceptions=False,
        )

    assert result.exit_code == 1, result.output
    assert result.output == snapshot


@mock.patch(
    "snowflake.cli.plugins.snowpark.commands._check_if_all_defined_integrations_exists"
)
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_procedure_fails_if_object_exists_and_no_replace(
    mock_object_manager,
    _,
    runner,
    mock_cursor,
    project_directory,
    snapshot,
):
    mock_object_manager.return_value.describe.return_value = mock_cursor(
        [
            ("packages", "[]"),
            ("handler", "hello"),
            ("returns", "string"),
        ],
        columns=["key", "value"],
    )

    with project_directory("snowpark_procedures"):
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ]
        )

    assert result.exit_code == 1
    assert result.output == snapshot


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_procedure_replace_nothing_to_update(
    mock_object_manager,
    mock_conn,
    runner,
    mock_cursor,
    mock_ctx,
    project_directory,
):
    mock_object_manager.return_value.describe.side_effect = [
        mock_cursor(
            [
                ("packages", "[]"),
                ("handler", "hello"),
                ("returns", "string"),
            ],
            columns=["key", "value"],
        ),
        mock_cursor(
            [
                ("packages", "[]"),
                ("handler", "test"),
                ("returns", "string"),
            ],
            columns=["key", "value"],
        ),
    ]
    ctx = mock_ctx()
    mock_conn.return_value = ctx

    with project_directory("snowpark_procedures"):
        result = runner.invoke(["snowpark", "deploy", "--replace", "--format", "json"])

    assert result.exit_code == 0, result.output
    assert json.loads(result.output) == [
        {
            "object": "procedureName(name string)",
            "status": "packages updated",
            "type": "procedure",
        },
        {"object": "test()", "status": "packages updated", "type": "procedure"},
    ]


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_procedure_replace_updates_single_object(
    mock_object_manager,
    mock_conn,
    runner,
    mock_cursor,
    mock_ctx,
    project_directory,
):
    mock_object_manager.return_value.describe.side_effect = [
        mock_cursor(
            [
                ("packages", "[]"),
                ("handler", "hello"),
                ("returns", "string"),
            ],
            columns=["key", "value"],
        ),
        mock_cursor(
            [
                ("packages", "[]"),
                ("handler", "foo"),
                ("returns", "string"),
            ],
            columns=["key", "value"],
        ),
    ]
    ctx = mock_ctx()
    mock_conn.return_value = ctx

    with project_directory("snowpark_procedures"):
        result = runner.invoke(["snowpark", "deploy", "--replace", "--format", "json"])

    assert result.exit_code == 0
    assert json.loads(result.output) == [
        {
            "object": "procedureName(name string)",
            "status": "packages updated",
            "type": "procedure",
        },
        {"object": "test()", "status": "definition updated", "type": "procedure"},
    ]


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_procedure_replace_creates_missing_object(
    mock_object_manager,
    mock_conn,
    runner,
    mock_cursor,
    mock_ctx,
    project_directory,
):
    mock_object_manager.return_value.describe.side_effect = [
        mock_cursor(
            [
                ("packages", "[]"),
                ("handler", "hello"),
                ("returns", "string"),
            ],
            columns=["key", "value"],
        ),
        ProgrammingError("does not exist or not authorized"),
    ]
    ctx = mock_ctx()
    mock_conn.return_value = ctx

    with project_directory("snowpark_procedures"):
        result = runner.invoke(["snowpark", "deploy", "--replace", "--format", "json"])

    assert result.exit_code == 0
    assert json.loads(result.output) == [
        {
            "object": "procedureName(name string)",
            "status": "packages updated",
            "type": "procedure",
        },
        {"object": "test()", "status": "created", "type": "procedure"},
    ]


@mock.patch("snowflake.connector.connect")
def test_execute_procedure(mock_connector, runner, mock_ctx):
    ctx = mock_ctx()
    mock_connector.return_value = ctx
    result = runner.invoke(
        [
            "snowpark",
            "execute",
            "procedure",
            "procedureName(42, 'string')",
        ]
    )

    assert result.exit_code == 0, result.output
    assert ctx.get_query() == "call procedureName(42, 'string')"


@mock.patch(
    "snowflake.cli.api.commands.project_initialisation._create_project_template"
)
def test_init_procedure(mock_create_project_template, runner, temp_dir):
    runner.invoke(["snowpark", "init", "my_project2"])
    mock_create_project_template.assert_called_once_with(
        "default_snowpark", project_directory="my_project2"
    )
