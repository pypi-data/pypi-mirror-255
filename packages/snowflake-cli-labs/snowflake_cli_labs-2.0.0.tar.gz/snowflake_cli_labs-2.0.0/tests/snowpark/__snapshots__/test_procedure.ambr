# serializer version: 1
# name: test_deploy_procedure_fails_if_integration_does_not_exists
  '''
  ╭─ Error ──────────────────────────────────────────────────────────────────────╮
  │ Following external access integration does not exists in Snowflake:          │
  │ external_2                                                                   │
  ╰──────────────────────────────────────────────────────────────────────────────╯
  
  '''
# ---
# name: test_deploy_procedure_fails_if_object_exists_and_no_replace
  '''
  ╭─ Error ──────────────────────────────────────────────────────────────────────╮
  │ Following objects already exists. Consider using --replace.                  │
  │ procedure: procedureName(string)                                             │
  │ procedure: test()                                                            │
  ╰──────────────────────────────────────────────────────────────────────────────╯
  
  '''
# ---
# name: test_deploy_procedure_secrets_without_external_access
  '''
  ╭─ Error ──────────────────────────────────────────────────────────────────────╮
  │ procedureName defined with secrets but without external integration.         │
  ╰──────────────────────────────────────────────────────────────────────────────╯
  
  '''
# ---
